examples/script_mode_train_any_tf_script_in_sage_maker.ipynb

```diff
@@ -0,0 +1,374 @@
+    "- The IAM role arn used to give training and hosting access to your data. See the documentation [for how to create these](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html). Note, if more than one role is required for notebook instances, training, and/or hosting, please replace the ```sagemaker.get_execution_role()``` with a the appropriate full IAM role arn string(s).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "bucket = sagemaker_session.default_bucket()\n",
+    "\n",
+    "role = sagemaker.get_execution_role()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Clone the repository"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!git clone https://github.com/sherjilozair/char-rnn-tensorflow > /dev/null 2>&1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This repository includes a README.md with an overview of the project, requirements, and basic usage:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import display, Markdown, Latex\n",
+    "display(Markdown('char-rnn-tensorflow/README.md'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Getting the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir sherlock\n",
+    "!wget https://sherlock-holm.es/stories/plain-text/cnus.txt --force-directories --output-document=sherlock/input.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload the data for training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = sagemaker_session.upload_data(path='sherlock', bucket=bucket, key_prefix='datasets/sherlock')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Testing locally"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Script Mode is still in developement phase. We will have to construct a Estimator to be able to use it with [SageMaker Python SDK](https://github.com/aws/sagemaker-python-sdk)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "from sagemaker.estimator import Framework\n",
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "class ScriptModeTensorFlow(Framework):\n",
```

Comment:
Current design is to add a flag to the existing TensorFlow estimator, script_mode=true. I talked to yuan briefly last week, we are going to deprioritize the sdk change until we get some feedbacks from the beta customers.

Beta users can use the current TensorFlow estimator with the script-mode containers. They just have to specify the entire image uri and ignore the extra arguments we passed in as hyperparamters, eg, training_steps and eval_steps. I think we can explain all of that here.

Reply:
The script in this example breaks with the TF estimator because the additional hyperparameters, which is why I had to create the estimator.

Reply:
I see. I think Nadia's idea of use scriptmode with py3 might work. This is fine as is and we can modify this notebook once we made the changes to the sdk.